### PR TITLE
Add complete support for new redis, fix tests and coverage

### PR DESCRIPTION
**Pull request by DaAwesomeP**
This pull:
 - New tests, 100% coverage (if you enable Coveralls)
 - Update to new `node_redis` version
 - Update other dependencies, remove `supertest` and `blanket`
 - Add support for all options in `node_redis` while keeping backward compatibility
 - Add support for the `duplicate` option
 - Add support for `quit()` and `end()` from the object
 - Add lots of details in docs (README)
 - Adds `hiredis` as an optional dependency (faster parser). Because it is optional, if it fails to build it will not cause any harm (it will use the JS parser as it was before). This is why Travis now has GCC.

@dead-horse I did not update the benchmarks, which most certainly have changed. It would also be nice if you could release a new version `v1.1.0` with this. This is backwards compatible with previous versions (see [lines 41-42](https://github.com/DaAwesomeP/koa-redis/blob/32a120f6886b9a5ae6661932c5e7a2d13cb04434/index.js#L41-L42), [71](https://github.com/DaAwesomeP/koa-redis/blob/32a120f6886b9a5ae6661932c5e7a2d13cb04434/index.js#L71)) despite the new way that options are passed to `node_redis`.

This squashes #12 (sorry @JacksonTian).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,44 +1,14 @@
...
-		--timeout $(TIMEOUT) \
-		--require should \
-		$(MOCHA_OPTS) \
-		$(TESTS)
+test-only:
+	NODE_ENV=test npm run-script test-only
 
 autod:
-	./node_modules/.bin/autod -w -e example --prefix=~ --keep=supertest,debug, --semver=koa@1
+	npm i autod && autod -w -e example --prefix=~ --keep=debug, --semver=koa@1
```
Comment (dead-horse) [inline Makefile]: make `autod` as a devDependence and move it in to `package.json`, we can remove `Makefile`

File: package.json [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
+  "keywords": [
+    "koa",
+    "middleware",
+    "session",
+    "redis"
+  ],
   "author": "dead_horse <dead_horse@qq.com>",
   "dependencies": {
-    "co-redis": "~1.1.0",
+    "co-redis": "2.0.*",
```
Comment (dead-horse) [inline package.json]: `~2.0.0`

File: test/koa-redis.test.js [hunk 3]
```diff
@@ -13,48 +13,119 @@
...
+      yield store.destroy('key:badKey');
       should.not.exist(yield store.get('key:nottl'));
       should.not.exist(yield store.get('key:ttl'));
+      should.not.exist(yield store.get('key:badKey'));
+      yield store.quit();
       done();
     })();
   });
 
   it('should expire after 1s', function (done) {
```
Comment (dead-horse) [inline test/koa-redis.test.js]: use `co-mocha`.

```js
it('should expire after 1s', function* () {
  yield store.set('key:ttl2', {a: 1, b: 2}, 1000);
  yield sleep(1000);
  yield store.quit();
});
```

File: Makefile (hunk 1 above)
Comment (DaAwesomeP) [inline Makefile]: Fixed in https://github.com/DaAwesomeP/koa-redis/commit/94e0b15344d4ea45b0e037ee9eafa84497721049

File: package.json (hunk 2 above)
Comment (DaAwesomeP) [inline package.json]: Fixed in https://github.com/DaAwesomeP/koa-redis/commit/a64752f72381b4a9a6d0a404c8969bf3bc39c6b2

File: test/koa-redis.test.js (hunk 3 above)
Comment (DaAwesomeP) [inline test/koa-redis.test.js]: I didn't know that that existed! Will do.

Comment (DaAwesomeP) [inline test/koa-redis.test.js]: @dead-horse How do you suggest I do the first and second tests that have events in them? SHould I just let them use `co`?

Comment (DaAwesomeP) [inline test/koa-redis.test.js]: @dead-horse Nevermind. I made a little listener-to-promise function. Fixed in https://github.com/DaAwesomeP/koa-redis/commit/8837c621fa82094845bb7543ea5cb15a3e99e06b.

**Comment by DaAwesomeP** [comment]
@dead-horse `co-mocha` added a really odd delay in every test. Sometimes tests would finish ~50ms later, and I even had to adjust sleep time in the ttl test (I tried to find an exact number, but it turned out to be pretty random). Since this problem was introduced by the tests themselves which were fine before, I don't think that it's a problem.

**Comment by DaAwesomeP** [comment]
You can see this problem in Travis [here](https://travis-ci.org/koajs/koa-redis/jobs/100985212#L306).

**Comment by dead-horse** [comment]
Hi, what's your npm user name? I'll add you as an onwer of this module.

**Comment by DaAwesomeP** [comment]
It's the same: `daawesomep`. Wow. Thank you so much.

**Comment by dead-horse** [comment]
added

**Pull request merged**